handler/src/main/java/io/netty/handler/ssl/OpenSslSessionCache.java

```diff
@@ -0,0 +1,493 @@
+                // The session is not valid anymore, let's remove it and just signal back that there is no session
+                // with the given ID in the cache anymore. This also takes care of calling SSL_SESSION_free(...)
+                removeSessionWithId(session.sessionId());
+                return null;
+            }
+            return session;
+        }
+    }
+
+    /**
+     * Returns a snapshot of the session ids of the current valid sessions.
+     */
+    final List<OpenSslSessionId> getIds() {
+        final OpenSslSession[] sessionsArray;
+        synchronized (this) {
+            sessionsArray = sessions.values().toArray(EMPTY_SESSIONS);
+        }
+        List<OpenSslSessionId> ids = new ArrayList<OpenSslSessionId>(sessionsArray.length);
+        for (OpenSslSession session: sessionsArray) {
+            if (session.isValid()) {
+                ids.add(session.sessionId());
+            }
+        }
+        return ids;
+    }
+
+    /**
+     * Clear the cache and free all cached SSL_SESSION*.
+     */
+    synchronized void clear() {
+        Iterator<Map.Entry<OpenSslSessionId, NativeSslSession>> iterator = sessions.entrySet().iterator();
+        while (iterator.hasNext()) {
+            NativeSslSession session = iterator.next().getValue();
+            iterator.remove();
+
+            // Notify about removal. This also takes care of calling SSL_SESSION_free(...).
+            notifyRemovalAndFree(session);
+        }
+    }
+
+    /**
+     * {@link OpenSslSession} implementation which wraps the native SSL_SESSION* while in cache.
+     */
+    static final class NativeSslSession implements OpenSslSession {
+        static final ResourceLeakDetector<NativeSslSession> LEAK_DETECTOR = ResourceLeakDetectorFactory.instance()
+                .newResourceLeakDetector(NativeSslSession.class);
+        private final ResourceLeakTracker<NativeSslSession> leakTracker;
+        private final long session;
+        private final String peerHost;
+        private final int peerPort;
+        private final OpenSslSessionId id;
+        private final long timeout;
+        private final long creationTime = System.currentTimeMillis();
+        private volatile long lastAccessedTime = creationTime;
+        private volatile boolean valid = true;
+        private boolean freed;
+
+        NativeSslSession(long session, String peerHost, int peerPort, long timeout) {
+            this.session = session;
+            this.peerHost = peerHost;
+            this.peerPort = peerPort;
+            this.timeout = timeout;
+            this.id = new OpenSslSessionId(io.netty.internal.tcnative.SSLSession.getSessionId(session));
+            leakTracker = LEAK_DETECTOR.track(this);
+        }
+
+        @Override
+        public void setSessionId(OpenSslSessionId id) {
+            throw new UnsupportedOperationException();
+        }
+
+        boolean shouldBeSingleUse() {
+            assert !freed;
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Use of identifier that is a keyword in later versions of Java (NM_FUTURE_KEYWORD_USED_AS_IDENTIFIER)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-use-of-identifier-that-is-a-keyword-in-later-versions-of-java-nm-future-keyword-used-as-identifier).
The identifier is a word that is reserved as a keyword in later versions of Java, and your code will need to be changed in order to compile it in later versions of Java.

Reply:
What identifier is problematic here?